[expo-audio] IllegalStateException when using more than one useAudioRecorderState

### Minimal reproducible example

https://github.com/vipierozan99/expo-bug-report/tree/expo-audio-race-cond

### Steps to reproduce

1. Open the app on the android emulator
2. Click the record button, Wait for a few seconds and press the stop recording button
3. After a few tries you should get a: 
```
Error: Call to function 'AudioRecorder.getStatus' has been rejected.
→ Caused by: java.lang.IllegalStateException, js engine: hermes
```

### Expected Behaviour

I would expect the app not to crash if there are multiple instances of `useAudioRecorderState` calling `AudioRecorder.getStatus`

Seems like a race condition on the native side, I believe [this](https://github.com/expo/expo/issues/37289) issue has the same underlying cause.

These are the offending lines:

https://github.com/vipierozan99/expo-bug-report/blob/d4d09ef95991844d23dd72f07f7d5905f4b6ffba/components/Recorder.tsx#L53-L61



### Environment

```text
expo-env-info 1.3.4 environment info:
  System:
    OS: Linux 6.15 Fedora Linux 41 (Workstation Edition)
    Shell: 5.2.32 - /bin/bash
  Binaries:
    Node: 23.6.0 - ~/.nvm/versions/node/v23.6.0/bin/node
    Yarn: 1.22.22 - ~/.nvm/versions/node/v23.6.0/bin/yarn
    npm: 10.9.2 - ~/.nvm/versions/node/v23.6.0/bin/npm
    Watchman: 20250118.093600.0 - /usr/local/bin/watchman
  SDKs:
    Android SDK:
      API Levels: 26, 34, 35, 36
      Build Tools: 26.0.3, 34.0.0, 35.0.0, 35.0.1, 36.0.0
      System Images: android-26 | Google APIs Intel x86_64 Atom, android-34 | Google APIs ARM 64 v8a, android-34 | Google APIs Intel x86_64 Atom, android-35 | Google Play Intel x86_64 Atom
```

### Expo Doctor Diagnostics

```text
15/15 checks passed. No issues detected!
```

--- a/packages/expo-audio/CHANGELOG.md
+++ b/packages/expo-audio/CHANGELOG.md
@@ -19,6 +19,7 @@
 - Add shallow state comparison to prevent unnecessary re-renders in `useAudioRecorderState`. ([#38273](https://github.com/expo/expo/pull/38273) by [@alanjhughes](https://github.com/alanjhughes))
 - [web] Fixed broken player state when `replace` was called while audio was running, added missing `updateInterval` to web, and improved audio status event emitting. ([#38505](https://github.com/expo/expo/pull/38505) by [@hirbod](https://github.com/hirbod))
 - [android][iOS][web] Fix recordForDuration status updates and add new record() options API ([#38612](https://github.com/expo/expo/pull/38612) by [@hirbod](https://github.com/hirbod))
+- [android]: Handle exceptions when retrieving maxAmplitude from MediaRecorder ([#38690](https://github.com/expo/expo/pull/38690) by [@hirbod](https://github.com/hirbod))
 
 ### 💡 Others
 

--- a/packages/expo-audio/android/src/main/java/expo/modules/audio/AudioRecorder.kt
+++ b/packages/expo-audio/android/src/main/java/expo/modules/audio/AudioRecorder.kt
@@ -49,7 +49,15 @@ class AudioRecorder(
       return null
     }
 
-    val amplitude: Int = recorder?.maxAmplitude ?: 0
+    val amplitude: Int = try {
+      recorder?.maxAmplitude ?: 0
+    } catch (e: Exception) {
+      // MediaRecorder maxAmplitude can throw various exceptions:
+      // - IllegalStateException: invalid recorder state/race condition
+      // - RuntimeException: getMaxAmplitude failed (hardware/driver issues)
+      // We return 0 (silence) as fallback for any amplitude reading failure
+      0
+    }
     return if (amplitude == 0) {
       -160.0
     } else {

